File: IntegrationTests/Parser/TypeParsing/TypeParserTests/Constructor/ConstructorSample.cs
// Copyright 2012 rubicon informationstechnologie gmbh
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

using System;
using InjectionCop.Fragment;

namespace InjectionCop.IntegrationTests.Parser.TypeParsing.TypeParserTests.Constructor
{
  public class ConstructorSample : ParserSampleBase
  {
    public ConstructorSample ()
    {
    }

    public ConstructorSample (string nonFragmentParameter)
    {
      DummyMethod (nonFragmentParameter);
    }

    public ConstructorSample (string nonFragmentParameter, [Fragment ("ConstructorFragment")] string fragmentParameter)
    {
      DummyMethod (nonFragmentParameter);
      RequiresConstructorFragment (fragmentParameter);
    }

    public ConstructorSample (string nonFragmentParameter, [Fragment ("ConstructorFragment")] string fragmentParameter, string dummy)
    {
      DummyMethod (nonFragmentParameter + fragmentParameter + dummy);
      RequiresConstructorFragment (nonFragmentParameter);
    }

    public ConstructorSample (string nonFragmentParameter, [Fragment ("ConstructorFragment")] string fragmentParameter, string dummy1, string dummy2)
      : this (nonFragmentParameter, fragmentParameter)
    {
      DummyMethod (nonFragmentParameter + fragmentParameter + dummy1 + dummy2);
    }

    public ConstructorSample (string nonFragmentParameter, [Fragment ("ConstructorFragment")] string fragmentParameter, string dummy1, string dummy2, string dummy3)
      : this (nonFragmentParameter, nonFragmentParameter, dummy1)
    {
      DummyMethod (nonFragmentParameter);
      DummyMethod (fragmentParameter);
      DummyMethod (dummy1);
      DummyMethod (dummy2);
      DummyMethod (dummy3);
    }

    public void RequiresConstructorFragment ([Fragment ("ConstructorFragment")] string fragmentParameter)
    {
      DummyMethod (fragmentParameter);
    }
  }
}

File: IntegrationTests.VB/Parser/TypeParsing/TypeParserTests/Constructor/ConstructorSample.vb
Imports InjectionCop.Fragment
Imports System

Namespace Parser.TypeParsing.TypeParserTests.Constructor
  Public Class ConstructorSample
    Inherits ParserSampleBase

    Public Sub New()
    End Sub

    Public Sub New(nonFragmentParameter As String)
      MyBase.DummyMethod(nonFragmentParameter)
    End Sub

    Public Sub New(nonFragmentParameter As String, <Fragment("ConstructorFragment")> fragmentParameter As String)
      MyBase.DummyMethod(nonFragmentParameter)
      Me.RequiresConstructorFragment(fragmentParameter)
    End Sub

    Public Sub New(nonFragmentParameter As String, <Fragment("ConstructorFragment")> fragmentParameter As String, dummy As String)
      MyBase.DummyMethod(nonFragmentParameter + fragmentParameter + dummy)
      Me.RequiresConstructorFragment(nonFragmentParameter)
    End Sub

    Public Sub New(nonFragmentParameter As String, <Fragment("ConstructorFragment")> fragmentParameter As String, dummy1 As String, dummy2 As String)
      Me.New(nonFragmentParameter, fragmentParameter)
      MyBase.DummyMethod(nonFragmentParameter + fragmentParameter + dummy1 + dummy2)
    End Sub

    Public Sub New(nonFragmentParameter As String, <Fragment("ConstructorFragment")> fragmentParameter As String, dummy1 As String, dummy2 As String, dummy3 As String)
      Me.New(nonFragmentParameter, nonFragmentParameter, dummy1)
      MyBase.DummyMethod(nonFragmentParameter)
      MyBase.DummyMethod(fragmentParameter)
      MyBase.DummyMethod(dummy1)
      MyBase.DummyMethod(dummy2)
      MyBase.DummyMethod(dummy3)
    End Sub

    Public Sub RequiresConstructorFragment(<Fragment("ConstructorFragment")> fragmentParameter As String)
      MyBase.DummyMethod(fragmentParameter)
    End Sub
  End Class
End Namespace
